src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.e2e.spec.ts

```diff
@@ -75,35 +89,48 @@ void describe('EventStoreDBEventStore', () => {
           data: { percent: discount, couponId: uuid() },
         },
       ],
-      {
-        projections: [
-          async ({ events, streamVersion, streamId }) => {
-            const state = events.reduce(evolve, null);
-            if (state === null) return;
-            await shoppingCartShortInfo.insertOne({
-              streamId,
-              version: streamVersion,
-              ...state,
-            });
-          },
-        ],
-      },
+      { expectedStreamVersion: STREAM_DOES_NOT_EXIST },
     );
 
-    const doc = await shoppingCartShortInfo.findOne({
-      streamId: shoppingCartId,
+    const stream = await collection.findOne({ streamName });
+    assertIsNotNull(stream);
+    assertEqual('3', stream.metadata.streamPosition.toString());
+    assertDeepEqual(stream.projections[SHOPPING_CART_PROJECTION_NAME]?.short, {
+      productItemsCount: 20,
+      totalAmount: 54,
     });
+  });
 
-    assertIsNotNull(doc);
+  void it('should only return a subset of stream events based on expectedStreamVersion', async () => {
+    const productItem: PricedProductItem = {
+      productId: '123',
+      quantity: 10,
+      price: 3,
+    };
+    const discount = 10;
+    const shoppingCartId = uuid();
+    const streamName = toStreamName('shopping_cart', shoppingCartId);
 
-    const { _id, ...docWithoutId } = doc;
+    await eventStore.appendToStream<ShoppingCartEvent>(
+      streamName,
+      [
+        { type: 'ProductItemAdded', data: { productItem } },
+        { type: 'ProductItemAdded', data: { productItem } },
+        {
+          type: 'DiscountApplied',
+          data: { percent: discount, couponId: uuid() },
+        },
+      ],
+      { expectedStreamVersion: STREAM_DOES_NOT_EXIST },
+    );
 
-    assertDeepEqual(docWithoutId, {
-      streamId: shoppingCartId,
-      version: 3,
-      productItemsCount: 20,
-      totalAmount: 54,
+    const expectedStreamVersion = 2;
```

Comment:
**QUESTION:** shouldn't we have here version number 3? As we're adding 3 events. I think that we discussed to start counting stream position from 1.

Reply:
Correct I forgot to adjust this

Reply:
Actually I'm only using this to confirm we only receive that many events from the stream if passed as a filter

Reply:
Oh I see now, that makes more sense.